python/ray/tests/test_advanced.py

```diff
@@ -521,6 +521,43 @@ def method(self):
     assert ray.worker.global_worker.core_worker.object_exists(x_id)
 
 
+@pytest.mark.skipif(client_test_enabled(), reason="internal api")
+def test_future_resolution_skip_plasma(ray_start_cluster):
```

Comment:
Although normally I'm all for test coverage, in this case, I think it's actually fine to not have such a test since it is testing internal behavior. It should be good enough as long as the existing tests for borrowed objects pass.

If you want to test the behavior, maybe you could add a small C++ test for the FutureResolver?

Reply:
I totally agree, although I had assumed that some of the Python tests, such as [this one](https://github.com/ray-project/ray/blob/912d0cbbf912d5b52d6176155bdff02f504b657d/python/ray/tests/test_basic_2.py#L451-L465), dipped into testing internal `CoreWorker` behavior because the `CoreWorker` [isn't very unit-testable](https://github.com/ray-project/ray/issues/13268). I added this Python test so the behavior of the e2e interaction is covered, and if I hadn't, I wouldn't have noticed a bug in the implementation that caused objects to never be inlined in the `GetObjectStatus` reply, so I'm reluctant to remove it!

It doesn't look like there are any tests for the `FutureResolver` yet, and that wouldn't test the full interaction (handling the `GetObjectStatus` reply), but that does seem like something that should be added. I'll look at adding those tests.

Would it be all right if I added some `FutureResolver` tests but kept the Python test that covers the e2e system behavior?

Reply:
Ah yes, good point about the lack of CoreWorker unit tests. Okay, I'm fine keeping this! I'm not sure if there's much FutureResolver to test so feel free to add that or not.